src/code/z_actor.c

```diff
@@ -457,439 +1269,3558 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+            flag = 0;
+        } else {
+            s32 dmgFlags = acHitInfo->toucher.dmgFlags;
+
+            if (dmgFlags & 0x800) {
+                flag = 1;
+            } else if (dmgFlags & 0x1000) {
+                flag = 2;
+            } else {
+                flag = (dmgFlags & 0x2000) ? 0x20 : 0;
+            }
+        }
+
+        actor->dropFlag |= flag;
+    }
+}
+
+void func_800BE33C(Vec3f* arg0, Vec3f* arg1, Vec3s* arg2, s32 arg3) {
+    f32 xDiff = arg1->x - arg0->x;
+    f32 zDiff = arg1->z - arg0->z;
+    f32 yDiff = arg3 ? (arg1->y - arg0->y) : (arg0->y - arg1->y);
+
+    arg2->y = Math_FAtan2F(zDiff, xDiff);
+    arg2->x = Math_FAtan2F(sqrtf(SQ(xDiff) + SQ(zDiff)), yDiff);
+}
+
+void func_800BE3D0(Actor* actor, s16 angle, Vec3s* arg2) {
+    f32 sp44;
+    f32 sp40;
+    f32 sp3C;
+    f32 sp38;
+    f32 sp34;
+    f32 sp30;
+    f32 sp2C;
+    s32 pad[3];
+
+    if (actor->floorPoly != NULL) {
+        CollisionPoly* floorPoly = actor->floorPoly;
+
+        sp44 = COLPOLY_GET_NORMAL(floorPoly->normal.x);
+        sp40 = COLPOLY_GET_NORMAL(floorPoly->normal.y);
+        sp3C = COLPOLY_GET_NORMAL(floorPoly->normal.z);
+
+        sp38 = Math_SinS(angle);
+        sp34 = Math_CosS(angle);
+        arg2->x = (s16)-Math_Atan2S((-(sp44 * sp38) - (sp3C * sp34)) * sp40, 1.0f);
+
+        sp2C = Math_SinS(angle - 0x3FF7);
+        sp30 = Math_CosS(angle - 0x3FF7);
+        arg2->z = (s16)-Math_Atan2S((-(sp44 * sp2C) - (sp3C * sp30)) * sp40, 1.0f);
+    }
+}
+
+void func_800BE504(Actor* actor, ColliderCylinder* collider) {
+    if ((collider->info.acHitInfo->toucher.dmgFlags & 0x13820)) {
+        actor->world.rot.y = collider->base.ac->shape.rot.y;
+    } else {
+        actor->world.rot.y = Actor_YawBetweenActors(collider->base.ac, actor);
+    }
+}
+
+void func_800BE568(Actor* actor, ColliderSphere* collider) {
+    if (collider->info.acHitInfo->toucher.dmgFlags & 0x13820) {
+        actor->world.rot.y = collider->base.ac->shape.rot.y;
+    } else {
+        actor->world.rot.y = Actor_YawBetweenActors(collider->base.ac, actor);
+    }
+}
+
+void func_800BE5CC(Actor* actor, ColliderJntSph* collider, s32 arg2) {
+    if (collider->elements[arg2].info.acHitInfo->toucher.dmgFlags & 0x13820) {
+        actor->world.rot.y = collider->base.ac->shape.rot.y;
+    } else {
+        actor->world.rot.y = Actor_YawBetweenActors(collider->base.ac, actor);
+    }
+}
+
+s32 func_800BE63C(struct EnBox* chest) {
+    if ((chest->unk_1F1 == 5) || (chest->unk_1F1 == 6) || (chest->unk_1F1 == 7) || (chest->unk_1F1 == 8) ||
+        (chest->unk_1F1 == 0xC)) {
+        return true;
+    }
+    return false;
+}
+
+TexturePtr* D_801AEFA8[] = {
+    D_04091DE0,
+    D_04091FE0,
+    D_040921E0,
+    D_040923E0,
+};
+
+#ifdef NON_MATCHING
+// stack is one variable too big
+void func_800BE680(GlobalContext* globalCtx, Actor* actor, Vec3f* limbPos, s16 arg3, f32 arg4, f32 arg5, f32 arg6,
+                   u8 mode) {
+    if (arg6 > 0.001f) {
```

Comment:
So this is another function that I was really interested in a while back. There isn't any immediate docs suggestions, especially since it's still non-matching, but I would like to share what I know:

This came up while decomping the Gekko minibosses in both woodfall temple and great bay temple. It seems to be directly tied to this enum I made which seems to represent a more global enum for the `mode` of this function:

https://github.com/zeldaret/mm/blob/97425f68aef4a8fc773a5ef6fb5bd61b51d40f71/src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.h#L23-L28

Reading this code, it does seem like mode 0 is "thawing" based on the `NA_SE_EV_BURN_OUT` sfxId, but then I'm curious what's needed for none of these effects to draw. The other type of effects were tested for pametfrog so I do feel fairly confident in that.